src/synthesizer/abundances/abundance_patterns.py

```diff
@@ -252,28 +112,40 @@ def __init__(
 
         """
 
+        # basic element info
+        self.metals = elements.Elements().metals
+        self.non_metals = elements.Elements().non_metals
+        self.all_elements = elements.Elements().all_elements
+        self.alpha_elements = elements.Elements().alpha_elements
+        self.element_name = elements.Elements().name
+        self.atomic_mass = elements.Elements().atomic_mass
+
         # save all arguments to object
         self.metallicity = metallicity  # mass fraction in metals
         self.alpha = alpha
-        self.solar = solar
+        self.reference = reference
         # depletion on to dust
         self.depletion = depletion
         self.depletion_model = depletion_model
         self.depletion_scale = depletion_scale
 
         # If depletion model is provided as a string use this to extract the
         # class.
-        if isinstance(solar, str):
-            if solar in solar_abundance_patterns.available_patterns:
-                self.solar = getattr(solar_abundance_patterns, solar)
+        if isinstance(reference, str):
+            if reference in reference_abundance_patterns.available_patterns:
+                self.reference = getattr(reference_abundance_patterns,
+                                         reference)
             else:
-                raise exceptions.UnrecognisedOption("""Solar abundance pattern
-                not recognised!""")
+                raise exceptions.UnrecognisedOption("""Reference abundance
+                pattern not recognised!""")
+
+        # initialise class
+        self.reference = self.reference()
```

Comment:
This fails if the user passes an instantiated class.

Reply:
Fixed